src/app/ngrx/effects/castle.effects.spec.ts

```diff
@@ -199,27 +165,232 @@ describe('CastleEffects', () => {
+        podcastId: episodes[0].podcastId,
+        page: 1,
+        all: true
+      });
+      const success = new ACTIONS.CastleEpisodePageSuccessAction({
+        episodes,
+        page: 1,
+        total: episodes.length,
+        all: true
+      });
+      actions$.stream = hot('-a', { a: action });
+      const expected = cold('-r', { r: success });
+      expect(effects.loadEpisodePage$).toBeObservable(expected);
+    });
+
+    it('should load next page of episodes', () => {
+      const pageOfEpisodes = new Array(EPISODE_PAGE_SIZE).fill(episodes[0]);
+      const firstPageAction = new ACTIONS.CastleEpisodePageSuccessAction({
+        episodes: pageOfEpisodes,
+        page: 1,
+        total: pageOfEpisodes.length * 2,
+        all: true
+      });
+      const nextPageLoadAction = new ACTIONS.CastleEpisodePageLoadAction({
+        podcastId: episodes[0].podcastId,
+        page: 2,
+        all: true
+      });
+      actions$.stream = hot('-a', { a: firstPageAction });
+      const expected = cold('-r', { r: nextPageLoadAction });
+      expect(effects.loadNextEpisodePage$).toBeObservable(expected);
+    });
+
+    it('fails to load a page of episodes', () => {
+      const error = new Error('Whaaaa?');
+      castle.root.mock('prx:podcast', {id: podcast.id}).mockError('prx:episodes', error);
+      const action = new ACTIONS.CastleEpisodePageLoadAction({podcastId: podcast.id, page: 1});
+      const completion = new ACTIONS.CastleEpisodePageFailureAction({error});
+      actions$.stream = hot('-a', {a: action});
+      const expected = cold('-r', {r: completion});
+      expect(effects.loadEpisodePage$).toBeObservable(expected);
+    });
+
+    it('should load metrics on episode page success', () => {
+      spyOn(store, 'dispatch').and.callThrough();
+      const action = new ACTIONS.CastleEpisodePageSuccessAction({
+        episodes,
+        page: routerParams.episodePage,
+        total: episodes.length,
+        all: true
+      });
+      const episodeMetricsActions = episodes.map(e => {
+        return new ACTIONS.CastleEpisodeMetricsLoadAction({
+          podcastId: e.podcastId,
+          page: e.page,
+          guid: e.guid,
+          metricsType: routerParams.metricsType,
+          interval: routerParams.interval,
+          beginDate: routerParams.beginDate,
+          endDate: routerParams.endDate
+        });
+      });
+      const completionString = episodes.map((e, i) => `-${i}`);
+      const completion = {};
+      // create an object with indices (toString) as keys to each action for marble mapping
+      episodeMetricsActions.forEach((a, i) => {
+        completion[i.toString()] = a;
+      });
+
+      actions$.stream = hot('-a--', {a: action});
+      // Marble syntax: '()' to sync groupings
+      // When multiple events need to be in the same frame synchronously, parentheses are used to group those events.
+      // eg '-(-0-1)' events '0' and '1' will both be in frame 10
```

Comment:
a bit more complicated marble testing here since this action does a mergeMap to create multiple actions for loading each episode's metrics from a single episode page success action, the metrics load actions all come out in the same frame, so they need to be grouped with `()` in the expected result.